rhino/src/main/java/org/mozilla/javascript/BaseFunction.java

```diff
@@ -313,6 +323,52 @@ static boolean isApplyOrCall(IdFunctionObject f) {
         return false;
     }
 
+    private Object makeHasInstance() {
+        Context cx = Context.getCurrentContext();
+        ScriptableObject obj = null;
+
+        if (cx != null) {
+            Scriptable scope = this.getParentScope();
+            obj =
```

Comment:
It's great to use a Lambda function here, but since this class still inherits from "IdScriptableObject", you should be able to make it work by adding a case to the switch in the "exec" function for your new "SymbolId_hasInstance" case.